csharp/src/SeedLang/Runtime/Value.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace SeedLang.Runtime {
+  // The value type used in the SeedLang core components.
+  //
+  // Design consideration:
+  // 1) It's designed as a value type (struct) to avoid object creating frequently. It's wrapped by
+  //    a ValueWrapper and sent to the visualizer center when visualizers need be notified.
+  // 2) "in" keyword is used when passing Value as a parameter of functions to avoid copying.
+  // 3) "ref readonly" keywords are used when returning Value from a function to avoid copying.
+  internal readonly struct Value : IEquatable<Value> {
+    internal enum ValueType {
+      None,
+      Boolean,
+      Number,
+      String,
+      List,
+    }
+
+    private readonly ValueType _type;
+    private readonly double _number;
+    private readonly HeapObject _object;
+
+    private Value(bool value) {
+      _type = ValueType.Boolean;
+      _number = ValueHelper.BooleanToNumber(value);
+      _object = null;
+    }
+
+    private Value(double value) {
+      _type = ValueType.Number;
+      _number = value;
+      _object = null;
+    }
+
+    private Value(string value) : this(ValueType.String, HeapObject.String(value)) { }
+
+    private Value(ValueType type, HeapObject box) {
```

Comment:
box -> obj or heapObj

Here and the rest.

Reply:
Done